servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ProtocolConfig.java

```diff
@@ -60,6 +61,35 @@ default String alpnId() {
     @Nullable
     KeepAlivePolicy keepAlivePolicy();
 
+    /**
+     * Get the {@link Http2Settings} that provides a hint for the initial settings. Note that some settings may be
+     * ignored if not supported (e.g. push promise).
```

Comment:
Consider clarifying that we throw an exception instead of ignoring

Reply:
the behavior may vary depending upon the use site and implementation, some settings maybe OK to ignore ... I don't think we need to over specify exception usage (illegal argument exception may always be thrown).